snapcraft/storeapi/__init__.py

```diff
@@ -609,6 +612,20 @@ def close_channels(self, snap_id, channel_names):
                                    response.content))
             raise errors.StoreChannelClosingError(response)
 
+    def sign_developer_agreement(self, latest_tos_accepted=True):
+        auth = _macaroon_auth(self.conf)
+        data = {'latest_tos_accepted': latest_tos_accepted}
+        response = self.post(
+            'agreement/', data=json.dumps(data),
+            headers={'Authorization': auth,
+                     'Content-Type': 'application/json',
+                     'Accept': 'application/json'})
+
+        if not response.ok:
+            raise errors.DeveloperAgreementSignError(
+                'Failed to sign developer ToS.')
```

Comment:
Hmm, `response.ok` not being ok may not be an indication of failing to sign the developer agreement and a 503 instead